### PR TITLE
Don't validate opatch in a no-patch install

### PR DESCRIPTION
**Pull request by mfielding**
The `--no-patch` argument to `check-swlib.sh` and `install-oracle.sh` installs the base Oracle release without applying patches.  This argument ends up setting `oracle_rel` to `base`.  If not applying patches, `opatch` is unnecessary.  And without access to My Oracle Support, there's no way to download it.

This change simply ships the opatch check if `oracle_rel` is set to `base`

Testcase: https://gist.github.com/mfielding/2a3279abcc560e3e29180c92f09b7d22

```
$ ./check-swlib.sh --ora-swlib-bucket gs://mfielding-oracle-software --no-patch
Command used:
./check-swlib.sh --ora-swlib-bucket gs://mfielding-oracle-software --no-patch

Running with parameters from command line or environment variables:

ORA_EDITION=EE
ORA_RELEASE=base
ORA_SWLIB_BUCKET=gs://mfielding-oracle-software
ORA_VERSION=19.3.0.0.0

Found Ansible: ansible-playbook is /usr/bin/ansible-playbook

Running Ansible playbook: ansible-playbook -i localhost,   check-swlib.yml

PLAY [Set correct RDBMS release] *********************************************************************************************************************************************************

TASK [Verify that Ansible on control node meets the version requirements] ****************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "Ansible version is 2.10.8, continuing"
}

TASK [common : populate-vars | Show resolved release] ************************************************************************************************************************************
ok: [localhost] => {
    "oracle_rel": "base"
}

PLAY [Check swlib] ***********************************************************************************************************************************************************************

TASK [check-swlib : check-swlib | Validate base software release] ************************************************************************************************************************
ok: [localhost] => (item={'name': '19c_gi', 'version': '19.3.0.0.0', 'files': ['V982068-01.zip'], 'sha256sum': ['D668002664D9399CF61EB03C0D1E3687121FC890B1DDD50B35DCBE13C5307D2E'], 'md5sum': ['t8TGb4AfktFPqg15HM2nIQ==']})
ok: [localhost] => (item={'name': '19c_base', 'version': '19.3.0.0.0', 'edition': ['EE', 'SE2'], 'files': ['V982063-01.zip'], 'sha256sum': ['BA8329C757133DA313ED3B6D7F86C5AC42CD9970A28BF2E6233F3235233AA8D8'], 'md5sum': ['GFi9DSgcYPTdq9h7HCFKTw==']})

PLAY RECAP *******************************************************************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=8    rescued=0    ignored=0
```

### CODE REVIEW

**Comment by jcnars** [comment]
Can you pls document here for completeness what is the error if this fix was not in place?

Thanks

**Comment by mfielding** [comment]
The error before applying this patch (copied from https://gist.github.com/mfielding/2a3279abcc560e3e29180c92f09b7d22):

```
TASK [check-swlib : check-swlib | Validate opatch patches] *******************************************************************************************************************************
failed: [localhost] (item={'category': 'OPatch', 'release': '19.3.0.0.0', 'patchnum': '6880880', 'patchfile': 'p6880880_190000_Linux-x86-64.zip', 'md5sum': ''}) => {"ansible_loop_var": "item", "changed": false, "cmd": "set -o pipefail\ngsutil ls -L gs://mfielding-oracle-software/p6880880_190000_Linux-x86-64.zip |  awk '/Content-Length/ { print $2 }'\n", "delta": "0:00:01.210791", "end": "2023-01-26 17:23:34.012642", "failed_when_result": true, "item": {"category": "OPatch", "md5sum": "", "patchfile": "p6880880_190000_Linux-x86-64.zip", "patchnum": "6880880", "release": "19.3.0.0.0"}, "msg": "non-zero return code", "rc": 1, "start": "2023-01-26 17:23:32.801851", "stderr": "CommandException: One or more URLs matched no objects.", "stderr_lines": ["CommandException: One or more URLs matched no objects."], "stdout": "", "stdout_lines": []}
...ignoring

TASK [check-swlib : check-swlib | Report gsutil failures (patch)] ************************************************************************************************************************
failed: [localhost] (item={'changed': False, 'end': '2023-01-26 17:23:34.012642', 'stdout': '', 'cmd': "set -o pipefail\ngsutil ls -L gs://mfielding-oracle-software/p6880880_190000_Linux-x86-64.zip |  awk '/Content-Length/ { print $2 }'\n", 'failed': True, 'delta': '0:00:01.210791', 'stderr': 'CommandException: One or more URLs matched no objects.', 'rc': 1, 'invocation': {'module_args': {'creates': None, 'executable': '/bin/bash', '_uses_shell': True, 'strip_empty_ends': True, '_raw_params': "set -o pipefail\ngsutil ls -L gs://mfielding-oracle-software/p6880880_190000_Linux-x86-64.zip |  awk '/Content-Length/ { print $2 }'\n", 'removes': None, 'argv': None, 'warn': True, 'chdir': None, 'stdin_add_newline': True, 'stdin': None}}, 'start': '2023-01-26 17:23:32.801851', 'msg': 'non-zero return code', 'stdout_lines': [], 'stderr_lines': ['CommandException: One or more URLs matched no objects.'], 'failed_when_result': True, 'item': {'category': 'OPatch', 'release': '19.3.0.0.0', 'patchnum': '6880880', 'patchfile': 'p6880880_190000_Linux-x86-64.zip', 'md5sum': ''}, 'ansible_loop_var': 'item'}) => {"ansible_loop_var": "item", "changed": false, "item": {"ansible_loop_var": "item", "changed": false, "cmd": "set -o pipefail\ngsutil ls -L gs://mfielding-oracle-software/p6880880_190000_Linux-x86-64.zip |  awk '/Content-Length/ { print $2 }'\n", "delta": "0:00:01.210791", "end": "2023-01-26 17:23:34.012642", "failed": true, "failed_when_result": true, "invocation": {"module_args": {"_raw_params": "set -o pipefail\ngsutil ls -L gs://mfielding-oracle-software/p6880880_190000_Linux-x86-64.zip |  awk '/Content-Length/ { print $2 }'\n", "_uses_shell": true, "argv": null, "chdir": null, "creates": null, "executable": "/bin/bash", "removes": null, "stdin": null, "stdin_add_newline": true, "strip_empty_ends": true, "warn": true}}, "item": {"category": "OPatch", "md5sum": "", "patchfile": "p6880880_190000_Linux-x86-64.zip", "patchnum": "6880880", "release": "19.3.0.0.0"}, "msg": "non-zero return code", "rc": 1, "start": "2023-01-26 17:23:32.801851", "stderr": "CommandException: One or more URLs matched no objects.", "stderr_lines": ["CommandException: One or more URLs matched no objects."], "stdout": "", "stdout_lines": []}, "msg": "ERROR locating OPatch patch p6880880_190000_Linux-x86-64.zip: ['CommandException: One or more URLs matched no objects.']"}
```

File: roles/check-swlib/tasks/main.yml [hunk 1]
```diff
@@ -62,7 +62,7 @@
   ignore_errors: true
   with_items:
     - "{{ opatch_patches }}"
-  when: item.release == oracle_ver
+  when: item.release == oracle_ver and oracle_rel != "base"
```
Comment (jcnars) [inline roles/check-swlib/tasks/main.yml:65]: Clarification sought:

What is the differnce between L48 and the change in L65?

To be consistent with line: https://github.com/google/bms-toolkit/blob/master/roles/check-swlib/tasks/main.yml#L48
```
  when: item.base == oracle_ver and item.release == oracle_rel
```

can we just arrange the proposed change so both are equally skipped from checks with a `--no-patch` invocation.

Comment (mfielding) [reply roles/check-swlib/tasks/main.yml:65]: The difference is that `opatch_patches` is set unconditionally (https://github.com/google/bms-toolkit/blob/743971aafde6813b9f9df6be98d2ee9696e335f2/roles/common/defaults/main.yml#L220) but that `gi_patches` and `rdbms_patches` are only set if there are patches to apply (or, in other words, if it's not the base release).

**Review by alex-basinov — APPROVED** [review]

**Pull request merged**